server/eventlog/eventlog.go

```diff
@@ -143,18 +149,58 @@ func GetEventLogChunk(ctx context.Context, env environment.Env, req *elpb.GetEve
 }
 
 func NewEventLogWriter(ctx context.Context, b interfaces.Blobstore, invocationId string) *EventLogWriter {
+	cw := chunkstore.New(
+		b,
+		&chunkstore.ChunkstoreOptions{
+			WriteBlockSize:       defaultLogChunkSize,
+			WriteTimeoutDuration: defaultChunkTimeout,
+			NoSplitWrite:         true,
+		},
+	).Writer(ctx, getEventLogPathFromInvocationId(invocationId))
 	return &EventLogWriter{
-		chunkstore.New(
-			b,
-			&chunkstore.ChunkstoreOptions{
-				WriteBlockSize:       defaultLogChunkSize,
-				WriteTimeoutDuration: defaultChunkTimeout,
-				NoSplitWrite:         true,
-			},
-		).Writer(ctx, getEventLogPathFromInvocationId(invocationId)),
+		WriteCloser: &ANSICursorBufferWriter{
+			WriteCloser:  cw,
+			screenWriter: terminal.NewScreenWriter(),
+		},
+		ChunkstoreWriter: cw,
 	}
 }
 
 type EventLogWriter struct {
-	*chunkstore.ChunkstoreWriter
+	io.WriteCloser
+	ChunkstoreWriter *chunkstore.ChunkstoreWriter
+}
+
+func (w *EventLogWriter) Write(p []byte) (int, error) {
+	return w.WriteCloser.Write(p)
+}
+
+type ANSICursorBufferWriter struct {
```

Comment:
can you add comments to this class explaining what it does / why?

Reply:
Yup, done.